testdata/cheats/CaptureLogs.t.sol

```diff
@@ -0,0 +1,137 @@
+// SPDX-License-Identifier: Unlicense
+pragma solidity >=0.8.0;
+
+import "forge-std/Test.sol";
+import "ds-test/test.sol";
+import "./Cheats.sol";
+
+contract Emitter {
+    event LogTopic0(
+        bytes data
+    );
+
+    event LogTopic1(
+        uint256 indexed topic1,
+        bytes data
+    );
+
+    event LogTopic12(
+        uint256 indexed topic1,
+        uint256 indexed topic2,
+        bytes data
+    );
+
+    event LogTopic123(
+        uint256 indexed topic1,
+        uint256 indexed topic2,
+        uint256 indexed topic3,
+        bytes data
+    );
+
+    function emitEvent(
+        bytes memory data
+    ) public {
+        emit LogTopic0(data);
+    }
+
+    function emitEvent(
+        uint256 topic1,
+        bytes memory data
+    ) public {
+        emit LogTopic1(topic1, data);
+    }
+
+    function emitEvent(
+        uint256 topic1,
+        uint256 topic2,
+        bytes memory data
+    ) public {
+        emit LogTopic12(topic1, topic2, data);
+    }
+
+    function emitEvent(
+        uint256 topic1,
+        uint256 topic2,
+        uint256 topic3,
+        bytes memory data
+    ) public {
+        emit LogTopic123(topic1, topic2, topic3, data);
+    }
+}
+
+contract CaptureLogsTest is DSTest {
+    Cheats constant cheats = Cheats(HEVM_ADDRESS);
+    Emitter emitter;
+    bytes constant lorem = "Lorem ipsum dolor sit amet, consectetur adipiscing elit";
+
+    function setUp() public {
+        emitter = new Emitter();
+    }
+
+    function generateTestData(uint8 n) internal pure returns (bytes memory) {
+        bytes memory output = new bytes(n);
+
+        for (uint8 i = 0; i < n; i++) {
+            output[i] = bytes1( uint8( uint16(i + 1) * 23 ) );
+        }
+
+        return output;
+    }
+
+    function testNoCaptureGet() public {
+        emitter.emitEvent(1, 2, 3, generateTestData(100));
+        (bytes32[] memory topics, bytes memory data) = cheats.nextCapturedLog();
```

Comment:
I wonder if this cheatcode should actually revert if there is no log, since I would assume you would only ever call it if you expect a log to be present

Reply:
Can be done.Yes.